etcdmain/etcd.go

```diff
@@ -62,6 +62,10 @@ func Main() {
 
 	var stopped <-chan struct{}
 
+	if cfg.name != defaultName {
```

Comment:
I think we need to make sure that the --initial-cluster flag has not been set, since name is not required.